contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol

```diff
@@ -431,6 +434,143 @@ contract ERC20BridgeSampler is
         }
     }
 
+    /// @dev Sample sell quotes from an arbitrary on-chain liquidity provider.
+    ///      Calls the provider contract with the default gas limit (200k).
+    /// @param providerAddress Address of the liquidity provider contract.
+    /// @param takerToken Address of the taker token (what to sell).
+    /// @param makerToken Address of the maker token (what to buy).
+    /// @param takerTokenAmounts Taker token sell amount for each sample.
+    /// @return makerTokenAmounts Maker amounts bought at each taker token
+    ///         amount.
+    function sampleSellsFromLiquidityProvider(
+        address providerAddress,
+        address takerToken,
+        address makerToken,
+        uint256[] memory takerTokenAmounts
+    )
+        public
+        view
+        returns (uint256[] memory makerTokenAmounts)
+    {
+        return sampleSellsFromLiquidityProvider(
+            providerAddress,
+            takerToken,
+            makerToken,
+            takerTokenAmounts,
+            DEFAULT_CALL_GAS
+        );
+    }
+
+    /// @dev Sample sell quotes from an arbitrary on-chain liquidity provider.
+    /// @param providerAddress Address of the liquidity provider contract.
+    /// @param takerToken Address of the taker token (what to sell).
+    /// @param makerToken Address of the maker token (what to buy).
+    /// @param takerTokenAmounts Taker token sell amount for each sample.
+    /// @param callGasLimit Gas limit to use for calls to the provider's
+    ///        `getSellQuote` method.
+    /// @return makerTokenAmounts Maker amounts bought at each taker token
+    ///         amount.
+    function sampleSellsFromLiquidityProvider(
+        address providerAddress,
+        address takerToken,
+        address makerToken,
+        uint256[] memory takerTokenAmounts,
+        uint256 callGasLimit
+    )
+        public
+        view
+        returns (uint256[] memory makerTokenAmounts)
+    {
+        uint256 numSamples = takerTokenAmounts.length;
+        makerTokenAmounts = new uint256[](numSamples);
+        for (uint256 i = 0; i < numSamples; i++) {
+            (bool didSucceed, bytes memory resultData) =
+                providerAddress.staticcall.gas(callGasLimit)(
+                    abi.encodeWithSelector(
+                        ILiquidityProvider(0).getSellQuote.selector,
+                        takerToken,
+                        makerToken,
+                        takerTokenAmounts[i]
+                    ));
+            uint256 buyAmount = 0;
```

Comment:
Nit - `makerTokenAmounts[i]` will be initialized to all zeros. Can just use `makerTokenAmounts[i] = abi.decode(..);`